packages/app/hooks/useQuestions.ts

```diff
@@ -0,0 +1,33 @@
+import { API } from "@template/api-client";
+import { ListQuestionsResponse } from "@template/common";
+import useSWR, { responseInterface } from "swr";
+
+const TEN_SECONDS_IN_MS = 100000;
+const FIFTEEN_SECOND_IN_MS = 150000;
+
+type questionsResponse = responseInterface<ListQuestionsResponse, any>;
+
+interface UseQuestionReturn {
+  questions: questionsResponse["data"];
+  questionsError: questionsResponse["error"];
+  mutateQuestions: questionsResponse["mutate"];
+}
+
+export function useQuestions(qid: number): UseQuestionReturn {
+  const {
+    data: questions,
+    error: questionsError,
+    mutate: mutateQuestions,
+  } = useSWR(
+    qid && `/api/v1/queues/${qid}/questions`,
+    async () => API.questions.index(Number(qid)),
+    {
+      refreshInterval: Math.floor(
+        Math.random() * (FIFTEEN_SECOND_IN_MS - TEN_SECONDS_IN_MS + 1) +
```

Comment:
Could we simplify this math to 
`Math.random() * (FIVE_SECONDS_IN_MS + 1) + TEN_SECONDS_IN_MS `

Reply:
i actually like Eddy's version more, since it preseves the shape of the interval formula, though it could probably also all be pulled out into a tiny util function. 

Let's not bikeshed over this tho. Probably fine to leave as is

Reply:
ohh I see it now. Sounds good!